tools/c7n_azure/c7n_azure/query.py

```diff
@@ -278,5 +264,26 @@ def get_session(self):
 
         return self._session
 
+    def enumerate_resources(self, parent_resource, type_info, **params):
```

Comment:
By moving this in here we can easily override it in resources which need custom enumeration (non-arm).